File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/RepoRule.java
package xyz.wingio.plugins.morehighlight;

import android.content.Context;

import com.aliucord.PluginManager;
import com.discord.simpleast.core.parser.ParseSpec;
import com.discord.simpleast.core.parser.Parser;
import com.discord.simpleast.core.node.Node;
import com.discord.simpleast.core.parser.Rule;
import com.discord.utilities.textprocessing.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
public final class RepoRule extends Rule<MessageRenderContext, LinkNode<MessageRenderContext>,MessageParseState> {
    private Context context;
    public RepoRule(Pattern pattern, Context context) {
        super(pattern);
        this.context = context;
    }

    @Override
    public ParseSpec<MessageRenderContext, MessageParseState> parse(Matcher matcher, Parser<MessageRenderContext, ? super LinkNode<MessageRenderContext>, MessageParseState> parser, MessageParseState s) {
        LinkNode textNode = new LinkNode(matcher.group(1) + "/" + matcher.group(2), String.format("https://github.com/%s/%s", matcher.group(1), matcher.group(2)), context);
        return new ParseSpec<>(textNode, s);
    }
}

File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/rule/RepoRule.kt
package xyz.wingio.plugins.morehighlight.rule

import android.content.Context
import com.aliucord.Utils
import com.discord.simpleast.core.parser.ParseSpec
import com.discord.simpleast.core.parser.Parser
import com.discord.simpleast.core.parser.Rule
import com.discord.utilities.textprocessing.MessageParseState
import com.discord.utilities.textprocessing.MessageRenderContext
import xyz.wingio.plugins.morehighlight.node.ClickableNode
import java.util.regex.Matcher
import java.util.regex.Pattern

class RepoRule(
    private val context: Context
): Rule<MessageRenderContext, ClickableNode<MessageRenderContext?>, MessageParseState>(
    Pattern.compile("^<gh:([A-Za-z0-9-]{1,39})/([A-Za-z0-9-]{1,39})>")
) {

    override fun parse(
        matcher: Matcher,
        parser: Parser<MessageRenderContext, in ClickableNode<MessageRenderContext?>?, MessageParseState>,
        s: MessageParseState
    ): ParseSpec<MessageRenderContext, MessageParseState> {
        val url = "https://github.com/%s/%s".format(matcher.group(1), matcher.group(2))

        val clickableNode = ClickableNode<MessageRenderContext>(matcher.group(1)!! + "/" + matcher.group(2), context, { Utils.launchUrl(url) }, {})
        return ParseSpec(clickableNode, s)
    }

}